src/middlewared/middlewared/etc_files/rc.conf.py

```diff
@@ -0,0 +1,457 @@
+        ('type', '=', 'RC'), ('enabled', '=', True)
+    ]):
+        yield f'{tun["var"]}="{tun["value"]}"'
+    return []
+
+
+def vmware_config(middleware, context):
+    try:
+        subprocess.run(
+            ['vmware-checkvm'],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+            check=True,
+        )
+    except subprocess.CalledProcessError:
+        yield 'vmware_guestd_enable="NO"'
+    except Exception:
+        middleware.logger.warn('Failed to run vmware-checkvm', exc_info=True)
+        return []
+    else:
+        yield 'vmware_guestd_enable="YES"'
+
+
+def _bmc_watchdog_is_broken():
+    cp = subprocess.run(['ipmitool', 'mc', 'info'], capture_output=True, errors='ignore')
+    reg = RE_FIRMWARE_VERSION.search(cp.stdout)
+    if not reg:
+        return False
+
+    version = reg.group(1).split('.')
+    if len(version) > 2:
+        return False
+
+    try:
+        return [int(i) for i in version[:2]] < [0, 30]
+    except ValueError:
+        return False
+
+
+def watchdog_config(middleware, context):
+    if context['is_freenas']:
+        # Bug #7337 -- blacklist AMD systems for now
+        model = sysctl.filter('hw.model')
+        if not model or 'AMD' not in model[0].value:
+            product = subprocess.run(
+                ['dmidecode', '-s', 'baseboard-product-name'],
+                capture_output=True,
+                errors='ignore',
+            ).stdout.split('\n')[0].strip()
+
+            if product in ('C2750D4I', 'C2550D4I') and _bmc_watchdog_is_broken():
+                return [
+                    'watchdogd_enable="YES"',
+                    'watchdogd_flags="-t 30 --softtimeout --softtimeout-action log,printf '
+                    '--pretimeout 15 --pretimeout-action log,printf -e \'sleep 1\' -w -T 3"',
+                ]
+            elif product not in ('X9DR3-F', 'X9DR3-LN4F+'):
+                return [
+                    'watchdogd_enable="YES"',
+                    'watchdogd_flags="--pretimeout 5 --pretimeout-action log,printf"',
+                ]
+    return ['watchdogd_enable="NO"']
+
+
+def zfs_config(middleware, context):
+    if middleware.call_sync('datastore.query', 'storage.volume'):
+        yield 'zfs_enable="YES"'
+
+
+def render(service, middleware):
+
+    context = get_context(middleware)
+
+    rcs = []
+    for i in (
+        services_config,
+        asigra_config,
+        collectd_config,
+        geli_config,
+        host_config,
+        kbdmap_config,
+        ldap_config,
+        lldp_config,
+        nfs_config,
+        nis_config,
+        nut_config,
+        powerd_config,
+        s3_config,
+        smart_config,
+        snmp_config,
+        staticroute_config,
+        tftp_config,
+        truenas_config,
+        tunable_config,
+        vmware_config,
+        watchdog_config,
+        zfs_config,
+    ):
+        rcs += list(i(middleware, context))
```

Comment:
Can we wrap it with `try ... except`? That's how it worked with bash I guess and if one service breaks, it won't drown the entire system